esp/platform/espp.cpp

```diff
@@ -452,7 +452,18 @@ int init_main(int argc, const char* argv[])
         else
         {
             envpt.setown(createPTreeFromXMLFile(cfgfile, ipt_caseInsensitive));
+
+            // NB: esp has no standard component config in bare-metal, this is loading defaultYaml only
             espConfig.setown(loadConfiguration(defaultYaml, argv, "esp", "ESP", nullptr, nullptr));
+
+            // legacy esp.xml will contain a generated global section if present in the environment.
+            // replace the empty stub created by loadConfiguration with this environment globals section.
+            Owned<IPropertyTree> global = envpt->queryPropTree("global");
+            if (global)
+            {
+                Owned<IPropertyTree> currentConfig = getComponentConfig();
+                replaceComponentConfig(currentConfig, global);
```

Comment:
The replaceComponentConfig made me confused. I thought it only replace the ComponentConfig(). I guess that the replaceComponentConfig means replacing the config for an HPCC component (including its GlobalConfig and its ComponentConfig.

Reply:
yes, the method can't replace one without replacing the other, so here Ionly want to replace the global, so am passing  the existing component config and the 'new' global (from env).